static/app/components/modals/projectCreationModal.tsx

```diff
@@ -0,0 +1,252 @@
+import {Fragment, useCallback, useState} from 'react';
+import {css} from '@emotion/react';
+import styled from '@emotion/styled';
+import omit from 'lodash/omit';
+import {PlatformIcon} from 'platformicons';
+
+import {
+  addErrorMessage,
+  addLoadingMessage,
+  addSuccessMessage,
+  clearIndicators,
+} from 'sentry/actionCreators/indicator';
+import type {ModalRenderProps} from 'sentry/actionCreators/modal';
+import {Button} from 'sentry/components/button';
+import Input from 'sentry/components/input';
+import type {SupportedLanguages} from 'sentry/components/onboarding/frameworkSuggestionModal';
+import PlatformPicker, {type Platform} from 'sentry/components/platformPicker';
+import TeamSelector from 'sentry/components/teamSelector';
+import {t} from 'sentry/locale';
+import ProjectsStore from 'sentry/stores/projectsStore';
+import {space} from 'sentry/styles/space';
+import type {OnboardingSelectedSDK, Organization, Team} from 'sentry/types';
+import {trackAnalytics} from 'sentry/utils/analytics';
+import slugify from 'sentry/utils/slugify';
+import useApi from 'sentry/utils/useApi';
+import type {IssueAlertFragment} from 'sentry/views/projectInstall/createProject';
+import IssueAlertOptions from 'sentry/views/projectInstall/issueAlertOptions';
+
+type Props = ModalRenderProps & {
+  organization: Organization;
+};
+
+export default function ProjectCreationModal({Header, closeModal, organization}: Props) {
+  const [platform, setPlatform] = useState<OnboardingSelectedSDK | undefined>(undefined);
+  const [step, setStep] = useState(0);
+  const [alertRuleConfig, setAlertRuleConfig] = useState<IssueAlertFragment | undefined>(
+    undefined
+  );
+  const [projectName, setProjectName] = useState('');
+  const [team, setTeam] = useState<Team | undefined>(undefined);
+  const [creating, setCreating] = useState(false);
+  const api = useApi();
+
+  function handlePlatformChange(selectedPlatform: Platform | null) {
+    if (selectedPlatform) {
+      setPlatform({
+        ...omit(selectedPlatform, 'id'),
+        key: selectedPlatform.id,
+      });
+    }
+  }
+
+  const createProject = useCallback(async () => {
```

Comment:
This might be a bit nicer as two useMutations which have `onError` and `onSuccess` but it's not that big of a deal to do it this way either.